src/unix/solarish/mod.rs

```diff
@@ -420,6 +420,24 @@ s! {
         pub portnfy_user: *mut ::c_void,
     }
 
+    pub struct aio_result_t {
+        aio_return: ::ssize_t,
+        aio_errno: ::c_int,
+    }
+
+    pub struct aiocb {
+        pub aio_fildes: ::c_int,
+        pub aio_buf: *mut ::c_void,
+        pub aio_nbytes: ::size_t,
+        pub aio_offset: ::off_t,
+        pub aio_reqprio: ::c_int,
+        pub aio_sigevent: sigevent,
+        pub aio_lio_opcode: ::c_int,
+        _aio_resultp: ::aio_result_t,
```

Comment:
I think these should just be named as they are in the header file, without the leading underscore, and they should be `pub`.  I think it's broadly a mistake to have any non-public fields in these struct definitions because it ostensibly means you can't construct them or read them if needed for whatever reason.  I believe I have personally had to do terrible `unsafe` stuff to work around this before.

Reply:
Fixed by making all fields public.